plugins/discourse-local-dates/spec/system/local_dates_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+describe "Local dates", type: :system, js: true do
+  fab!(:topic) { Fabricate(:topic) }
+    fab!(:user) { Fabricate(:user) }
+    fab!(:post) { Fabricate(:post_with_local_dates, topic: topic) }
+
+    it "renders local dates and date ranges correctly" do
+      sign_in user
+
+      visit "/t/#{topic.id}"
+      topic_page = PageObjects::Pages::Topic.new
+
+      post_dates = topic_page.find_all("span[data-date]")
```

Comment:
Before doing this you may want to make sure you are on the right page by checking the page has the post content e.g. `expect(page).to have_content("Date range post test")` or something similar, since capybara will wait for that but it will not wait for `find_all`